Remove .catch(noop) to make errors in Promises be logged in the console

DanReyLop opened the pull request:
Fixes #556

The default behaviour (in Chrome) when a Promise is rejected and there is no `.catch` handler is to log the error in the console. So by removing the `.catch(noop)` the errors will be logged.

@allendav @nabsul @robobot3000 @jeffstieler

allendav commented:
Hmmm. I gave this issue kinda a bad title. It be better to replace the catch noop with a catch that presents an error in the form (and disables submittal or something) - that way the end user can report that error to us.

A lot of merchants will not know how / think to look in Dev Console and will just think the form is broken.

We should also log a detailed error in dev console for site devs and tech-savvy merchants.

DanReyLop commented:
Ok, so I made a few changes:
* Added error logging to all the REST endpoints. If the endpoint is going to return a `WP_Error`, then that error will be logged.
* Changed the `normalizeAddress` Promise slightly. Now, if the server returns an error, the Promise will **resolve** with a value of `false`, instead of being rejected. So now we can handle success (promise resolved), validation error (promise resolved with `false`), and catastrophic error (promise rejected).
* Every time a Promise is rejected, the error will be logged in the console and an Error Notice will appear with the errror. Once we solve the issue of "make the notices be *on top* of the dialog", this should be enough.

jeffstieler commented on classes/class-wc-rest-connect-address-normalization-controller.php:
```diff
@@ -65,12 +71,24 @@ public function update_items( $request ) {
 				$response->error->code,
 				$response->error->message,
 				array( 'message' => $response->error->message )
 			);
```
You could use [`WP_Error::add_data()`](https://developer.wordpress.org/reference/classes/wp_error/add_data/) here instead of creating a new `WP_Error`.

jeffstieler commented:
@DanReyLop can you list out your testing steps for this PR? Is there an easy way to get all the requests to trigger the error condition? (perhaps turning off a local server?)

DanReyLop commented:
@jeffstieler Yes, turning off the local server before a request is the easiest way to test this. Note that for some cases (such as purchasing a label) you need to have the server running to go through all the process, and turn it off before hitting the `Purchase` button.

jeffstieler commented:
I'm not crazy about our message for timeouts ("Error: The WooCommerce Connect server returned ( 0 ) and an empty response body."), but this looks good.
